contracts/pool/BlendedPool.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.16;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import "../interfaces/IPoolFactory.sol";
+
+import "../interfaces/IHeliosGlobals.sol";
+import "../library/PoolLib.sol";
+import "../token/PoolFDT.sol";
+import "./AbstractPool.sol";
+
+/// @title Blended Pool
+contract BlendedPool is AbstractPool {
+    using SafeMathUint for uint256;
+    using SafeCast for uint256;
+    using SafeERC20 for IERC20;
+
+    address public immutable superFactory; // The factory that deployed this Pool
+
+    uint256 public principalOut; // The sum of all outstanding principal on Loans. TODO ??
+    address public borrower; // Address of borrower for this Pool.
+    mapping(address => bool) public pools;
+
+    event RegPoolDeposit(address indexed regPool, uint256 amount);
+
+    constructor(
+        address _liquidityAsset,
+        address _llFactory,
+        uint256 _lockupPeriod,
+        uint256 _apy,
+        uint256 _duration,
+        uint256 _minInvestmentAmount,
+        uint256 _withdrawThreshold
+    ) AbstractPool(_liquidityAsset, _llFactory, PoolLib.NAME, PoolLib.SYMBOL) {
+        require(_liquidityAsset != address(0), "P:ZERO_LIQ_ASSET");
+        require(_llFactory != address(0), "P:ZERO_LIQ_LOCKER_FACTORY");
+
+        superFactory = msg.sender;
+
+        poolInfo = PoolInfo(
+            _lockupPeriod,
+            _apy,
+            _duration,
+            type(uint256).max,
+            _minInvestmentAmount,
+            _withdrawThreshold
+        );
+    }
+
+    function withdrawableOf(address _holder) external view returns (uint256) {
+        require(
+            depositDate[_holder] + poolInfo.lockupPeriod <= block.timestamp,
+            "P:FUNDS_LOCKED"
+        );
+
+        return
+            Math.min(
+                liquidityAsset.balanceOf(address(liquidityLocker)),
+                super.balanceOf(_holder)
+            );
+    }
+
+    function totalLA() external view returns (uint256) {
+        return liquidityLocker.totalBalance();
+    }
+
+    function totalDeposited() external view returns (uint256) {
+        return totalMinted;
+    }
+
+    function decimals() public view override returns (uint8) {
+        return uint8(liquidityAssetDecimals);
+    }
+
+    /// @notice Used to transfer the investor's rewards to him
+    function claimReward() external override returns (bool) {
+        uint256 callerRewards = rewards[msg.sender];
+        require(callerRewards >= 0, "P:NOT_HOLDER");
```

Comment:
Why does the error say "P:NOT_HOLDER"?

callerRewards may be equal to zero even if the investor is "holder", but has no rewards.